backend/pkg/api/connect/service/quota/v1/service.go

```diff
@@ -0,0 +1,169 @@
+		})
+		if err != nil {
+			return nil, connect.NewError(
+				connect.CodeInternal,
+				fmt.Errorf("failed to apply pagination: %w", err),
+			)
+		}
+		quotaEntries = page
+		nextPageToken = token
+	}
+
+	response := &v1.ListQuotasResponse{
+		Quotas:    quotaEntries,
+		PageToken: nextPageToken,
+	}
+
+	return connect.NewResponse(response), nil
+}
+
+// BatchSetQuota creates client throughput quotas for specified entities.
+func (s *Service) BatchSetQuota(ctx context.Context, req *connect.Request[v1.BatchSetQuotaRequest]) (*connect.Response[v1.BatchSetQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.batchSetQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to create quota: %w", err))
+	}
+
+	response := &v1.BatchSetQuotaResponse{}
+
+	return connect.NewResponse(response), nil
+}
+
+// DeleteQuota deletes client quotas for specified entities and value types.
+func (s *Service) DeleteQuota(ctx context.Context, req *connect.Request[v1.DeleteQuotaRequest]) (*connect.Response[v1.DeleteQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.deleteQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to delete quota: %w", err))
+	}
+
+	response := &v1.DeleteQuotaResponse{}
+
+	return connect.NewResponse(response), nil
+}
+
+// SetQuota creates or updates client throughput quotas for a single entity.
+func (s *Service) SetQuota(ctx context.Context, req *connect.Request[v1.SetQuotaRequest]) (*connect.Response[v1.SetQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.setQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to set quota: %w", err))
+	}
+
+	response := &v1.SetQuotaResponse{}
+
+	return connect.NewResponse(response), nil
+}
+
+// BatchDeleteQuota deletes client quotas for multiple entities and value types.
+func (s *Service) BatchDeleteQuota(ctx context.Context, req *connect.Request[v1.BatchDeleteQuotaRequest]) (*connect.Response[v1.BatchDeleteQuotaResponse], error) {
+	kafkaReq, err := s.kafkaClientMapper.batchDeleteQuotaRequestToKafka(req.Msg)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	_, err = s.consoleSvc.AlterClientQuotas(ctx, kafkaReq)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("failed to batch delete quota: %w", err))
+	}
+
+	response := &v1.BatchDeleteQuotaResponse{}
```

Comment:
We need to adapt the response here (same problem as with batch set) - inner kafka errors may exist (e.g. authz issues)